convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,109 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+import os
+
+from convert2rhel import actions, systeminfo
+from convert2rhel.systeminfo import system_info
+
+
+logger = logging.getLogger(__name__)
+
+FIREWALLD_CONFIG_FILE = "/etc/firewalld/firewalld.conf"
+
+
+def _check_for_modules_cleanup_config():
+    """Verify firewalld modules cleanup config
+
+    :rtype: bool
+    :returns: Whether or not the CleanupModulesOnExit is set to true in
+        firewalld config.
+    """
+    if os.path.exists(FIREWALLD_CONFIG_FILE):
+        contents = []
+        with open(FIREWALLD_CONFIG_FILE, mode="r") as handler:
+            contents = [line.strip() for line in handler.readlines() if line]
+
+        # Contents list is empty for some reason, better to assume that there
+        # is no content in the file that was read.
+        if not contents:
+            return False
+
+        # If the config file has this option set to true/yes, then we need to
+        # return True to ask the user to change it to self.
+        if "CleanupModulesOnExit=yes" in contents or "CleanupModulesOnExit=true" in contents:
```

Comment:
Use regex here to expect whitespaces.
```
"CleanupModulesOnExit\s*=\s*yes"
```

I've just tried on RHEL 8 that:
1. `CleanupModulesOnExit = yes` works ok (firewalld doesn't complain after service restart)
2. Multiline ('\n') does not work:
```
CleanupModulesOnExit=
  yes
```

Reply:
updated